### PR TITLE
Clean header files

### LINKED ISSUE #236: Move function definitions from header files to source files when possible
When it is possible (and sensible), move source code for definitions of functions from header files to source code files.

### PR DESCRIPTION
**Pull request by Adda0**
This PR does some late spring-cleaning of header files. It mainly moves definitions of multitude of functions into their respective source files. Furthermore, the PR fixes some documentation comments, code aligning, includes, file headers (as per the agreement from #102), conversion of tab indentation to spaces etc.

All changes are made to header files. Changes in source files only reflect the changes in header files.

The PR #249 modifies `include/number-predicate.hh` heavily, so refactorization of this header file will be done in a separate PR.

Lastly, the commit https://github.com/VeriFIT/mata/pull/250/commits/aa13e720430f73bf00e88e37844ea7cf22b99f2f brings changes from #226 so that #226 can be closed, and we do not need to refactor that whole PR.

I suggest that the reviewers review the PR with whitespace changes hidden to get a somewhat more readable diff. Whitespace changes can be easily turned off in `Files changed` window settings, as shown in, for example, [this article](https://cloudfour.com/thinks/quick-tip-how-to-hide-whitespace-changes-in-git-diffs/#github).

### CODE REVIEW

File: src/nfa/nfa.cc [hunk 1]
```diff
@@ -740,12 +828,12 @@ Nfa Mata::Nfa::remove_epsilon(const Nfa& aut, Symbol epsilon) {
     while (changed) { // compute the fixpoint
         changed = false;
         for (size_t i=0; i < num_of_states; ++i) {
-            const auto& state_transitions{ aut.delta[i] };
-            const auto state_symbol_transitions{
-                state_transitions.find(Move{epsilon}) };
-            if (state_symbol_transitions != state_transitions.end()) {
+            const Post& post{ aut.delta[i] };
+            const auto emove{
```
Comment (jurajsic) [inline src/nfa/nfa.cc]: I would rename `emove` to `eps_move`, so it is more obvious.

**Review by jurajsic — APPROVED** [review]

File: src/nfa/nfa.cc (hunk 1 above)
Comment (Adda0) [reply src/nfa/nfa.cc]: Done.

**Comment by Adda0** [comment]
Issues from @jurajsic's review have been addressed, as well as what we discussed earlier.

File: include/mata/mintermization.hh [hunk 2]
```diff
@@ -1,9 +1,6 @@
-/*
- * mintermization.hh -- Mintermization of automaton
+/* mintermization.hh -- Mintermization of automaton.
  * It transforms an automaton with a bitvector formula used as a symbol to minterminized version of the automaton.
  *
- * Copyright (c) 2022 Martin Hruska <hruskamartin25@gmail.com>
```
Comment (tfiedor) [inline include/mata/mintermization.hh:5]: :D Yeah remove any mention of this bitch.

File: include/mata/nfa-algorithms.hh [hunk 3]
```diff
@@ -38,102 +36,100 @@
...
+ * @brief Compute intersection of two NFAs with a possibility of using multiple epsilons.
+ *
+ * @param[in] lhs First NFA to compute intersection for.
+ * @param[in] rhs Second NFA to compute intersection for.
+ * @param[in] preserve_epsilon Whether to compute intersection preserving epsilon transitions.
+ * @param[in] epsilons Set of symbols to be considered as epsilons
+ * @param[out] prod_map Mapping of pairs of the original states (lhs_state, rhs_state) to new product states.
+ * @return NFA as a product of NFAs @p lhs and @p rhs with ε-transitions preserved.
+ */
+Nfa intersection_eps(const Nfa& lhs, const Nfa& rhs, bool preserve_epsilon, const std::set<Symbol>& epsilons,
```
Comment (tfiedor) [inline include/mata/nfa-algorithms.hh:115]: Suggestion: Rename to `intersection_with_eps`?

File: include/mata/nfa-algorithms.hh [hunk 4]
```diff
@@ -38,102 +36,100 @@
...
+ * Supports epsilon symbols when @p use_epsilon is set to true.
+ * @param[in] lhs First automaton to concatenate.
+ * @param[in] rhs Second automaton to concatenate.
+ * @param[in] epsilon Epsilon to be used co concatenation (provided @p use_epsilon is true)
+ * @param[in] use_epsilon Whether to concatenate over epsilon symbol.
+ * @param[out] lhs_result_states_map Map mapping lhs states to result states.
+ * @param[out] rhs_result_states_map Map mapping rhs states to result states.
+ * @return Concatenated automaton.
+ */
+Nfa concatenate_eps(const Nfa& lhs, const Nfa& rhs, const Symbol& epsilon, bool use_epsilon = false,
```
Comment (tfiedor) [inline include/mata/nfa-algorithms.hh:130]: Suggestion: `concatenate_with_eps`?

File: include/mata/nfa.hh [hunk 5]
```diff
@@ -791,21 +591,14 @@ public:
...
 
-        //he, what is this? Some comment would help.
-        // I am thinking about that removing everything having to do with Transition might be a good thing. Transition adds clutter and makes people write inefficient code.
-        void refresh_trans()
-        { // {{{
-            this->trans = {trIt, this->tlIt->symbol, *(this->ssIt)};
-        } // }}}
+        // FIXME: He, what is this? Some comment would help.
+        // I am thinking about that removing everything having to do with Transition might be a good thing. Transition
+        //  adds clutter and makes people write inefficient code.
```
Comment (tfiedor) [inline include/mata/nfa.hh:596]: Is this even used? This looks like some very weird shit.

File: include/mata/ord-vector.hh [hunk 6]
```diff
@@ -1,12 +1,17 @@
-/*****************************************************************************
- *  Mata Tree Automata Library
+/* ord-vector.hh -- Implementation of a set (ordered vector) using std::vector.
  *
- *  Copyright (c) 2011  Ondra Lengal <ilengal@fit.vutbr.cz>
```
Comment (tfiedor) [inline include/mata/ord-vector.hh:4]: Yeah! Remove that bitch!

**Review by tfiedor — APPROVED** [review]
Seems fine. There are some functions that could be potential moved from header (especially in `closed-sets.hh`. But for first refactoring of header it is acceptable.

**Comment by Adda0** [comment]
> There are some functions that could be potential moved from header (especially in closed-sets.hh.

They are template classes and functions. They cannot be moved from the header, sadly.

File: include/mata/mintermization.hh (hunk 2 above)
Comment (Adda0) [reply include/mata/mintermization.hh:5]: As agreed in #102.

File: include/mata/nfa-algorithms.hh (hunk 3 above)
Comment (Adda0) [reply include/mata/nfa-algorithms.hh:115]: That can be done, but not for this PR. I do not want any more changes which actually change anything in the interface. Will add this to my todo list.

Comment (tfiedor) [reply include/mata/nfa-algorithms.hh:115]: Yes, this also should be discussed with others (probably).

File: include/mata/nfa.hh (hunk 5 above)
Comment (Adda0) [reply include/mata/nfa.hh:596]: No idea. I will not make any changes to the interface here, but I will take a note and revisit this later.

File: include/mata/nfa-algorithms.hh (hunk 4 above)
Comment (Adda0) [reply include/mata/nfa-algorithms.hh:130]: I will make a note and resolve it later, as explain in the conversation about `intersection_with_eps`.

File: include/mata/nfa.hh (hunk 5 above)
Comment (tfiedor) [reply include/mata/nfa.hh:596]: I think there is something like "Find usage" in IDEs. or `grep -r "refresh_transition" .`

Comment (Adda0) [reply include/mata/nfa.hh:596]: Yes, the problem here is that I have no idea whether it is meant to be used somewhere some day. As of now, it has no current usages.

File: include/mata/nfa-algorithms.hh (hunk 3 above)
Comment (Adda0) [reply include/mata/nfa-algorithms.hh:115]: I believe that the function was already named as you suggest, but then some objected that the name is too long.

File: include/mata/ord-vector.hh (hunk 6 above)
Comment (Adda0) [reply include/mata/ord-vector.hh:4]: An update of file headers is in the works.

**Pull request merged**